eppy/tests/test_function_helpers.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import pytest
+from six import StringIO
+
+import eppy.function_helpers as fh
+from eppy.bunch_subclass import register_epbunch_function
+from eppy.iddcurrent import iddcurrent
+from eppy.modeleditor import IDF
+from eppy.tests.test_bunch_subclass import idftxt
+
+
+class TestRegisterFunction:
+    @pytest.fixture()
+    def IDF(self):
+        iddfhandle = StringIO(iddcurrent.iddtxt)
+
+        if IDF.getiddname() == None:
+            IDF.setiddname(iddfhandle)
+        yield IDF
+
+    @pytest.fixture()
+    def idf(self, IDF):
+        idfhandle = StringIO(idftxt)
+        yield IDF(idfhandle)
+
+    def test_register_function_area(self, idf):
+        """Add a function that comes default with eppy"""
+
+        @register_epbunch_function(
+            "new_area",
+            keys=[
+                "BuildingSurface:Detailed",
+                "Wall:Detailed",
+                "RoofCeiling:Detailed",
+                "Floor:Detailed",
+                "FenestrationSurface:Detailed",
+                "Shading:Site:Detailed",
+                "Shading:Building:Detailed",
+                "Shading:Zone:Detailed",
+            ],
+        )
+        @property
+        def area(abunch):
+            return fh.area(abunch)
+
+        surf = idf.idfobjects["BuildingSurface:Detailed".upper()][0]
+        assert surf.area
+
+    def test_register_function_conditioned_area(self, idf):
+        """Add a new custom function that does not come default with eppy"""
+
+        @register_epbunch_function("conditioned_area", keys=["Zone"])
+        @property
+        def conditioned_area(abunch):
+            zone = abunch
+            area = 0
+            for surface in zone.zonesurfaces:
+                if surface.tilt == 180.0:
+                    part_of = int(zone.Part_of_Total_Floor_Area.upper() != "NO")
```

Comment:
No need to cast to int. A boolean is equivalent to 1 or 0 and can be used in calculations already.

It should also be named `is_part_of`. Just a personal preference thing, but booleans should call attention to the fact that they're booleans, with names like `is_something` or `has_something`.

Reply:
corrected in 48ae29efb22b7b34e8311c27ef8e5a89bc627cd8